zerver/tests/test_message_edit.py

```diff
@@ -907,6 +916,140 @@ def test_edit_cases(self) -> None:
+        )
+        self.assert_json_success(result_3)
+
+        result_4 = self.client_patch(
+            f"/json/messages/{msg_id}",
+            {
+                "topic": "topic 3",
+                "content": "content 3",
+            },
+        )
+        self.assert_json_success(result_4)
+
+        result_5 = self.client_patch(
+            f"/json/messages/{msg_id}",
+            {
+                "topic": "topic 4",
+                "stream_id": stream_3.id,
+            },
+        )
+        self.assert_json_success(result_5)
+
+        do_set_realm_property(
+            user.realm,
+            "message_edit_history_visibility",
+            MessageEditHistoryVisibilityEnum.MESSAGE_EDIT_HISTORY_VISIBILITY_NONE,
+            acting_user=None,
+        )
+        result_message_edit_history_none = self.client_get(f"/json/messages/{msg_id}/history")
+        self.assert_json_error(
+            result_message_edit_history_none,
+            "Message edit history is disabled in this organization",
+        )
+
+        do_set_realm_property(
+            user.realm,
+            "message_edit_history_visibility",
+            MessageEditHistoryVisibilityEnum.MESSAGE_EDIT_HISTORY_VISIBILITY_MOVES_ONLY,
+            acting_user=None,
+        )
+        result_message_edit_history_moves_only = self.client_get(f"/json/messages/{msg_id}/history")
+        json_response = orjson.loads(result_message_edit_history_moves_only.content)
+        message_edit_history_moves_only = json_response["message_history"]
+
+        for edit_history_entry in message_edit_history_moves_only:
+            self.assertNotIn("prev_content", edit_history_entry)
+            self.assertNotIn("prev_rendered_content", edit_history_entry)
+            self.assertNotIn("content_html_diff", edit_history_entry)
+
+        do_set_realm_property(
+            user.realm,
+            "message_edit_history_visibility",
+            MessageEditHistoryVisibilityEnum.MESSAGE_EDIT_HISTORY_VISIBILITY_ALL,
+            acting_user=None,
+        )
+        result_message_edit_history_all = self.client_get(f"/json/messages/{msg_id}/history")
+        json_response = orjson.loads(result_message_edit_history_all.content)
+        message_edit_history_all = json_response["message_history"]
+
+        for i, entry in enumerate(message_edit_history_all):
+            expected_entries = {"content", "rendered_content", "topic", "timestamp", "user_id"}
+            if i in {2, 4, 5}:
+                expected_entries.add("prev_topic")
+                expected_entries.add("topic")
+            if i in {1, 4}:
+                expected_entries.add("prev_content")
+                expected_entries.add("prev_rendered_content")
+                expected_entries.add("content_html_diff")
+            if i in {3, 5}:
+                expected_entries.add("prev_stream")
+                expected_entries.add("stream")
+            self.assertEqual(expected_entries, set(entry.keys()))
```

Comment:
I am not sure having these checks is helpful. Would be just better to have a check for length and to verify the values in those entries.